src/overlays/actors/ovl_En_Fr/z_en_fr.c

```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+    if (this->reward == 0) {
+        this->actionFunc = EnFr_Idle;
+    } else {
+        this->actionFunc = EnFr_GiveReward;
+        func_8002F434(&this->actor, globalCtx, this->reward, 30.0f, 100.0f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1C474.s")
+void EnFr_GiveReward(EnFr* this, GlobalContext* globalCtx) {
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        this->actor.parent = NULL;
+        this->actionFunc = EnFr_SetIdle;
+    } else {
+        func_8002F434(&this->actor, globalCtx, this->reward, 30.0f, 100.0f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1C524.s")
+void EnFr_SetIdle(EnFr* this, GlobalContext* globalCtx) {
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 6) && (func_80106BC8(globalCtx))) {
+        this->actionFunc = EnFr_Idle;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1C580.s")
+void EnFr_UpdateIdle(EnFr* this, GlobalContext* globalCtx) {
+    if (BREG(0)) {
+        DebugDisplay_AddObject(this->actor.posRot.pos.x, this->actor.posRot.pos.y, this->actor.posRot.pos.z,
+                               this->actor.posRot.rot.x, this->actor.posRot.rot.y, this->actor.posRot.rot.z, 1.0f, 1.0f,
+                               1.0f, 255, 0, 0, 255, 4, globalCtx->state.gfxCtx);
+    }
+    this->jumpCounter++;
+    this->actionFunc(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1C718.s")
+s32 EnFr_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, void* thisx,
+                          Gfx** gfx) {
+    if ((limbIndex == 7) || (limbIndex == 8)) {
+        *dList = NULL;
+    }
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1C790.s")
+void EnFr_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, void* thisx) {
+    EnFr* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1C7CC.s")
+    if ((limbIndex == 7) || (limbIndex == 8)) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1C830.s")
+        OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_fr.c", 1735);
+        Matrix_Push();
+        func_800D1FD4(&globalCtx->mf_11DA0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1C944.s")
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_en_fr.c", 1738),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_OPA_DISP++, *dList);
+        Matrix_Pull();
+        CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_fr.c", 1741);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fr/func_80A1CAAC.s")
+void EnFr_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s16 NoGlowRadius;
+    EnFr* this = THIS;
```

Comment:
id just call this lightRadius. the NoGlow in the name refers to the fact that the light doesnt have a glow halo thing around it